daemon/protocol/python/tests/run_many.py

```diff
@@ -12,174 +12,183 @@
-        self.assertCompleteStatuses(statuses, runs)
-            
-    
-    # make the following snapshots
-    # sleep snapshot: has a "sleep_script.sh" that sleeps for 0.5 seconds
-    # ls snapshot: has an "ls_script.sh" that issues a successful ls command
-    # ls fail snapshot: has an "ls_script.sh" that issues a failing ls command
-    # will return a dict of the snapshot ids
-    def make_snapshots(self):
-        # make the files that will go into the snapshots        
-        sleep_script = os.path.join(self.tmpdir, "sleep_script.sh")
-        open(sleep_script, 'w').write("#!/bin/sh\nsleep 0.5")
-        os.chmod(sleep_script, stat.S_IRUSR | stat.S_IWUSR | stat.S_IXUSR)
-    
-        ls_script = os.path.join(self.tmpdir, "ls_script.sh")
-        open(ls_script, 'w').write("#!/bin/sh\nls resource.txt")
-        os.chmod(ls_script, stat.S_IRUSR | stat.S_IWUSR | stat.S_IXUSR)
-    
-        resource = os.path.join(self.tmpdir, "resource.txt")
-        open(resource, 'w').write("content")
+tmpdir = None
+  
+def setUp():
+  global tmpdir
+  tmpdir = tempfile.mkdtemp()
+
+  proto.start()
+
+  #wait up to 1 second for daemon to start 
+  start = time.time()
+  elapsedTime = 0
+  started = False
+
+  while not started and elapsedTime < 1.0:    
+      try:
+          echo = proto.echo("ping")
+          if echo == "ping":
+              started = True
+      except proto.ScootException as e:
+          elapsedTime = time.time() - start
+  
+  if not started:
+      raise Exception("Connection to daemon couldn't be established in {0} seconds".format(elapsedTime))
+                    
+def tearDown():
+    shutil.rmtree(tmpdir)
+    proto.stop_daemon()
     
-        # make the snapshots
-        sleep_ss_id = proto.create_snapshot(sleep_script)
-        ls_ss_id = proto.create_snapshot(os.path.join(self.tmpdir, "*"))
-        fail_ls_ss_id = proto.create_snapshot(ls_script)
-        
-        # return a dict of the snapshot ids
-        return {self.sleep_ss_key:sleep_ss_id, self.ls_ss_key:ls_ss_id, self.ls_fail_ss_key:fail_ls_ss_id}
+sleep_ss_key = "sleep_s_id"
+ls_ss_key = "ls_s_id"
+ls_fail_ss_key = "ls_fail_s_id"
+
+# test scenario
+class CmdDef(object):
+    def __init__(self, cmd, snapshot_key, interm_states, final_state):
+        self.cmd = cmd
+        self.snapshot_key = snapshot_key
+        self.interm_states = interm_states
+        self.final_state = final_state
     
+sleepDef = CmdDef(cmd=["./sleep_script.sh"], 
+                  snapshot_key="sleep_s_id", 
+                  interm_states=[proto.ScootStatus.State.PENDING, 
+                                 proto.ScootStatus.State.PREPARING,
+                                 proto.ScootStatus.State.RUNNING, 
+                                 proto.ScootStatus.State.COMPLETED], 
+                  final_state=proto.ScootStatus.State.COMPLETED)  # run sleep 1, expect complete or running, final status:complete
```

Comment:
think these comments would be easier to read if they precede the instance object creation